libs/wire-api/src/Wire/API/Routes/Public.hs

```diff
@@ -176,27 +181,54 @@ type InternalAuth ztype opts =
     (ZHeader ztype)
     (ZParam ztype)
 
-type ZLocalUser = ZAuthServant 'ZLocalAuthUser InternalAuthDefOpts
+type ZLocalUser = ZAuthServant 'ZLocalAuthUser InternalAuthDefOpts 'Nothing
+
+type ZUser = ZAuthServant 'ZAuthUser InternalAuthDefOpts 'Nothing
+
+type ZClient = ZAuthServant 'ZAuthClient InternalAuthDefOpts 'Nothing
 
-type ZUser = ZAuthServant 'ZAuthUser InternalAuthDefOpts
+type ZConn = ZAuthServant 'ZAuthConn InternalAuthDefOpts 'Nothing
 
-type ZClient = ZAuthServant 'ZAuthClient InternalAuthDefOpts
+type ZBot = ZAuthServant 'ZAuthBot InternalAuthDefOpts 'Nothing
 
-type ZConn = ZAuthServant 'ZAuthConn InternalAuthDefOpts
+type ZConversation = ZAuthServant 'ZAuthConv InternalAuthDefOpts 'Nothing
 
-type ZBot = ZAuthServant 'ZAuthBot InternalAuthDefOpts
+type ZProvider = ZAuthServant 'ZAuthProvider InternalAuthDefOpts 'Nothing
 
-type ZConversation = ZAuthServant 'ZAuthConv InternalAuthDefOpts
+type ZOptUser = ZAuthServant 'ZAuthUser '[Servant.Optional, Servant.Strict] 'Nothing
 
-type ZProvider = ZAuthServant 'ZAuthProvider InternalAuthDefOpts
+type ZOptClient = ZAuthServant 'ZAuthClient '[Servant.Optional, Servant.Strict] 'Nothing
 
-type ZOptUser = ZAuthServant 'ZAuthUser '[Servant.Optional, Servant.Strict]
+type ZOptConn = ZAuthServant 'ZAuthConn '[Servant.Optional, Servant.Strict] 'Nothing
 
-type ZOptClient = ZAuthServant 'ZAuthClient '[Servant.Optional, Servant.Strict]
+type ZOAuthLocalUser (scopes :: [OAuthScope]) = ZAuthServant 'ZLocalAuthUser InternalAuthDefOpts ('Just scopes)
 
-type ZOptConn = ZAuthServant 'ZAuthConn '[Servant.Optional, Servant.Strict]
+type ZOauthUser (scopes :: [OAuthScope]) = ZAuthServant 'ZAuthUser InternalAuthDefOpts ('Just scopes)
 
-instance HasSwagger api => HasSwagger (ZAuthServant 'ZAuthUser _opts :> api) where
+instance
+  (HasSwagger api, IsOAuthScopes scopes, scopes ~ (s ': ss), Typeable ztype) =>
+  HasSwagger (ZAuthServant (ztype :: ZType) _opts ('Just scopes) :> api)
+  where
+  toSwagger _ =
+    toSwagger (Proxy @(ZAuthServant ztype _opts ('Nothing :: Maybe [OAuthScope]) :> api))
+      & securityDefinitions <>~ SecurityDefinitions (InsOrdHashMap.singleton "OAuth" secScheme)
+      & security <>~ [SecurityRequirement $ InsOrdHashMap.singleton "OAuth" []]
+      & addScopeDescription
+    where
+      secScheme =
+        SecurityScheme
+          { _securitySchemeType = SecuritySchemeApiKey (ApiKeyParams "Authorization" ApiKeyHeader),
+            _securitySchemeDescription =
+              Just $
+                "Must be a token retrieved with an oauth handshake. It must be presented in this \
+                \format: 'Bearer \\<token\\>'.\
+                \Further reading: https://docs.wire.com/how-to/install/oauth.html"
+          }
+
+      addScopeDescription :: Swagger -> Swagger
+      addScopeDescription = allOperations . description %~ Just . (<> "OAuth scope(s): " <> showOAuthScopeList @scopes) . fold
```

Comment:
swagger: scopes are there (cool!), but the other security information is not visible in swagger-ui, or at least i can't find it.  also, it's not even converted completely into swagger.json, or am i reading this wrong?

![2023-02-08-094841_1920x1080_scrot](https://user-images.githubusercontent.com/10210727/217480830-10ca0992-35a8-4aea-bd82-1fa0b08d9d94.png)
![2023-02-08-094844_1920x1080_scrot](https://user-images.githubusercontent.com/10210727/217480835-22868e45-c217-4b45-86d2-b8bc8cae5a0e.png)